src/core/libmaven/mzSample.cpp

```diff
@@ -899,7 +898,7 @@ void mzSample::populateFilterline(string filterLine, Scan *_scan)
 	}
 }
 
-void mzSample::parseMzXMLScan(const xml_node &scan, int scannum)
+void mzSample::parseMzXMLScan(const xml_node &scan, const int& scannum)
```

Comment:
@rish9511 Why pass int value as a const ref? pass-by-value would surely be same or faster in this case. Not really an issue, just curious.

Reply:
@shubhra-agrawal  
references make the code memory efficient not necessarily faster(unless it's a really large object)
if the parameters are just being read, they should be received as const references

Reply:
I get that. I was reading about passing by const ref recently and most sources were divided on what is better for native datatypes: sending by value or const ref. Just wanted to know why you chose one over the other. Seems you're going with convention here, which is good. Thanks!